docs/tutorials/customValidationModule.md

```diff
@@ -0,0 +1,267 @@
+    address smartAccount
+) public view virtual override returns (bytes4) {
+    if (
+        _verifySignature(
+            keccak256(
+                abi.encodePacked(
+                    "\x19Ethereum Signed Message:\n52",
+                    dataHash,
+                    smartAccount
+                )
+            ),
+            moduleSignature,
+            smartAccount
+        )
+    ) {
+        return EIP1271_MAGIC_VALUE;
+    }
+    return bytes4(0xffffffff);
+}
+```
+As per the code 
+- `isValidSignature`: this method validates an EIP-1271 signature. It internally calls the isValidSignatureForAddress which again calls the _verifySignature method.
+
+**4. Implement isValidSignatureUnsafe method**
+```typescript
+    /// @inheritdoc ISignatureValidator
+    function isValidSignatureUnsafe(
+        bytes32 dataHash,
+        bytes memory moduleSignature
+    ) public view virtual returns (bytes4) {
+        return
+            isValidSignatureForAddressUnsafe(
+                dataHash,
+                moduleSignature,
+                msg.sender
+            );
+    }
+
+    /// @inheritdoc IEcdsaOwnershipRegistryModule
+    function isValidSignatureForAddressUnsafe(
+        bytes32 dataHash,
+        bytes memory moduleSignature,
+        address smartAccount
+    ) public view virtual returns (bytes4) {
+        if (_verifySignature(dataHash, moduleSignature, smartAccount)) {
+            return EIP1271_MAGIC_VALUE;
+        }
+        return bytes4(0xffffffff);
+    }
+```
+As per the code
+- `isValidSignatureUnsafe`: this method validates an EIP-1271 signature but expects the data Hash to already include smart account address information.
+
+**5. Implement _verifySignature method**
+```typescript
+/**
+* @param dataHash Hash of the data to be validated.
+* @param signature Signature to be validated.
+* @param smartAccount expected signer Smart Account address.
+* @return true if signature is valid, false otherwise.
+*/
+function _verifySignature(
+    bytes32 dataHash,
+    bytes memory signature,
+    address smartAccount
+) internal view returns (bool) {
+
+    // verification according to signature scheme of this module
+}
+```
+As per the code `_verifySignature` method contains the custom logic for this module, which developer can implement based on their requirements.
+
+Expand the code below to see the entire code:
+<details>
+
+```javascript
+
+import {BaseAuthorizationModule} from "./BaseAuthorizationModule.sol";
+
+contract MyCustomValidationModule is BaseAuthorizationModule {
+    string public constant NAME = "My Custom Validation Module";
+    string public constant VERSION = "0.0.1";
+
+    /**
+        Smart contract account calls this function with msg.sender as smart account address
+        It updates the relevant storage for the msg sender, It could be ownership information as mentioned in ECDSAOwnsership Module.
+        Function signature of this method will be used as moduleSetupData in Account Factory
```

Comment:
```suggestion
        Smart contract account calls this function with msg.sender as smart account address
        It updates the relevant storage for the msg sender, It could be ownership information as mentioned in ECDSAOwnsership Module.
        The function signature of this method will be used as moduleSetupData in Account Factory
```